spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/GatewayMetricsAutoConfiguration.java

```diff
@@ -40,11 +43,18 @@
 		MetricsAutoConfiguration.class })
 public class GatewayMetricsAutoConfiguration {
 
+	@Bean
+	@ConditionalOnMissingBean(GatewayTagsProvider.class)
```

Comment:
You only need the `GatewayTagsProvider` in the annotation if you put `DefaultGatewayTagsProvider` as the return type.

Reply:
Fixed